examples/bri-3/src/bri/vsm/capabilites/executeVsmCycle/executeVsmCycleCommand.handler.ts

```diff
@@ -0,0 +1,58 @@
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { LoggingService } from '../../../../shared/logging/logging.service';
+import { TransactionStorageAgent } from '../../../transactions/agents/transactionStorage.agent';
+import { TransactionAgent } from '../../../transactions/agents/transactions.agent';
+import { TransactionStatus } from '../../../transactions/models/transactionStatus.enum';
+import { WorkstepStorageAgent } from '../../../workgroup/worksteps/agents/workstepsStorage.agent';
+import { ExecuteVsmCycleCommand } from './executeVsmCycle.command';
+
+@CommandHandler(ExecuteVsmCycleCommand)
+export class ExecuteVsmCycleCommandHandler
+  implements ICommandHandler<ExecuteVsmCycleCommand>
+{
+  constructor(
+    private agent: TransactionAgent,
+    private workstepStorageAgent: WorkstepStorageAgent,
+    private txStorageAgent: TransactionStorageAgent,
+    private readonly logger: LoggingService,
+  ) {}
+
+  async execute(command: ExecuteVsmCycleCommand) {
+    const executionCandidates =
+      await this.txStorageAgent.getTopNTransactionsByStatus(
+        Number(process.env.VSM_CYCLE_TX_BATCH_SIZE),
+        TransactionStatus.Initialized,
+      );
+
+    executionCandidates.forEach(async (tx) => {
+      tx.updateStatusToProcessing();
+      await this.txStorageAgent.updateTransactionStatus(tx);
+
+      if (!this.agent.validateTransactionForExecution(tx)) {
+        tx.updateStatusToInvalid();
+        await this.txStorageAgent.updateTransactionStatus(tx);
+        return;
+      }
+
+      const workstep = await this.workstepStorageAgent.getWorkstepById(
+        tx.workstepInstanceId,
+      );
+
+      try {
+        const txResult = this.agent.executeTransaction(tx, workstep!);
+        // TODO: When do we update the nonce on the BpiAccount?
```

Comment:
@ognjenkurtic after the transaction has been sent. For simplicity, the nonce is updated even if the transaction eventually fails. This ensures that no two nonces are used.

Reply:
Cool, nonce updates will come as part of #702